integrations/oci/oci-secrets-config-source/src/main/java/io/helidon/integrations/oci/secrets/configsource/OciSecretsConfigSourceProvider.java

```diff
@@ -0,0 +1,526 @@
+            try (Vaults v = vaultsSupplier.get()) {
+                return v.listSecrets(listSecretsRequest).getItems();
+            } catch (RuntimeException e) {
+                throw e;
+            } catch (Exception e) {
+                if (e instanceof InterruptedException) {
+                    Thread.currentThread().interrupt();
+                }
+                throw new IllegalStateException(e.getMessage(), e);
+            }
+        }
+
+        @SuppressWarnings("checkstyle:linelength")
+        private static ValueNode valueNode(Secrets s, SecretSummary ss, Base64.Decoder d) {
+            return
+                ValueNode.create(new String(d.decode(((Base64SecretBundleContentDetails) (s.getSecretBundle(GetSecretBundleRequest.builder()
+                                                                                                                .secretId(ss.getId())
+                                                                                                                .build())
+                                                                                              .getSecretBundle()
+                                                                                              .getSecretBundleContent()))
+                                                     .getContent()),
+                                            UTF_8)
+                                     .intern());
+        }
+
+
+        /*
+         * Inner and nested classes.
+         */
+
+
+        private static final class Builder extends AbstractConfigSourceBuilder<Builder, Void> {
+
+
+            /*
+             * Instance fields.
+             */
+
+
+            private String compartmentOcid;
+
+            private Supplier<? extends Secrets> secretsSupplier;
+
+            private String vaultOcid;
+
+            private Supplier<? extends Vaults> vaultsSupplier;
+
+
+            /*
+             * Constructors.
+             */
+
+
+            private Builder() {
+                super();
+                Supplier<? extends BasicAuthenticationDetailsProvider> adpSupplier =
+                    LazyValue.create(() -> (BasicAuthenticationDetailsProvider) ociAuthenticationProvider().get());
+                SecretsClient.Builder scb = SecretsClient.builder();
+                this.secretsSupplier = () -> scb.build(adpSupplier.get());
+                VaultsClient.Builder vcb = VaultsClient.builder();
+                this.vaultsSupplier = () -> vcb.build(adpSupplier.get());
+            }
+
+
+            /*
+             * Instance methods.
+             */
+
+
+            @Override // AbstractConfigSourceBuilder<Builder, Void>
+            protected Builder config(Config metaConfig) {
+                metaConfig.get("compartment-ocid")
+                    .asString()
+                    .flatMap(s -> s.isBlank() ? Optional.empty() : Optional.of(s))
```

Comment:
this can be replaced with `.filter(Predicate.not(String::isBlank))`

Reply:
Doh! I keep getting burned by the fact the `Config#asString()` method documentation says it returns an `Optional`. I will make this change.

Reply:
Done